### PR TITLE
fix: correct implementation of nextIterativeCombination method to handle singular input

### LINKED ISSUE #80: fix nextIterativeCombination method to correctly output combinations given a singular input
# Story
As a user of Combo
I need correct output of combinations given a singular combination input e.g `TEST1: "foo", "bar"` 
# Summary
We need to fix the bug in `nextIterativeCombination()` that is causing the output of the previous example to be `TEST1: "foo"` instead of the correct output `TEST1: "foo", TEST1: "bar"` This will ensure that `nextIterative` provides correct combinations. 
# Acceptance Criteria
* Correct output given the example test case
* Passes all tests

Note: This bug is only an issue in `nextIterativeCombination` and not `nextPreSolvedCombination`

### PR DESCRIPTION
**Pull request by jbuchananr**
This PR adds an edge case to `nextIterativeCombination()` method that correctly handles the case given an input of a single parameter

e.g
Input: `test: "foo", "bar"`
Output:
```---
test: foo
---
test: bar
```

closes #80

### CODE REVIEW

File: pkg/combination/combination.go [hunk 1]
```diff
@@ -86,6 +86,17 @@ func (cs *stream) nextIterativeCombination() (map[string]string, error) {
 	// comboList is a variable that holds a list of combinations to be returned.
 	comboList := map[string]string{}
 
+	// Edge case: 1 parameter
```
Comment (tylerslaton) [inline pkg/combination/combination.go:89]: This will work but from a readability stand-point this function is getting a bit out of hand. I'd like to see some comments and have the logic broken up a bit where makes sense. For example, a couple of these for loops could potentially go into other functions or refactored.

**Review by tylerslaton — CHANGES_REQUESTED** [review]
Good catch! I would like to see a unit test that checks for this new logic if you don't mind adding that.

File: pkg/combination/combination.go (hunk 1 above)
Comment (jbuchananr) [reply pkg/combination/combination.go:89]: I think that's a great idea. Would we want the readability improved in this PR or in a separate issue?

Comment (tylerslaton) [reply pkg/combination/combination.go:89]: Let's make it a follow-up.

**Review by tylerslaton — APPROVED** [review]
Look great! Let's be sure to improve the readability of the `nextIterative` function in #82

**Comment by exdx** [comment]
Looks like this closes #79?

Closes #79
